keanu-project/src/main/java/io/improbable/keanu/util/DescriptionCreator.java

```diff
@@ -0,0 +1,325 @@
+            }
+        }
+        return scalarValue;
+    }
+
+    private String getDescriptionDelimiter(Vertex<?> vertex) {
+        return delimiters.getOrDefault(vertex.getClass(), ", ");
+    }
+
+    private Optional<String> checkForIrregularExpressions(Vertex<?> vertex, boolean includeBrackets) {
+        if (vertex instanceof BooleanIfVertex || vertex instanceof DoubleIfVertex || vertex instanceof IntegerIfVertex) {
+            Optional<String> description = createIfStringDescription(vertex, includeBrackets);
+            if (description.isPresent()) {
+                return Optional.of(description.get());
+            }
+        } else if (vertex instanceof BooleanBinaryOpVertex) {
+            String booleanBinaryDescription = createBooleanBinaryOpDescription(
+                (BooleanBinaryOpVertex) vertex,
+                delimiters.getOrDefault(vertex.getClass(), ", "),
+                includeBrackets);
+            return Optional.of(booleanBinaryDescription);
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<String> createIfStringDescription(Vertex<?> vertex, boolean includeBrackets) {
+        BooleanVertex predicate;
+        Vertex<?> thn;
+        Vertex<?> els;
+
+        if (vertex instanceof IntegerIfVertex) {
+            predicate = ((IntegerIfVertex) vertex).getPredicate();
+            thn = ((IntegerIfVertex) vertex).getThn();
+            els = ((IntegerIfVertex) vertex).getEls();
+        } else if (vertex instanceof BooleanIfVertex) {
+            predicate = ((BooleanIfVertex) vertex).getPredicate();
+            thn = ((BooleanIfVertex) vertex).getThn();
+            els = ((BooleanIfVertex) vertex).getEls();
+        } else if (vertex instanceof DoubleIfVertex) {
+            predicate = ((DoubleIfVertex) vertex).getPredicate();
+            thn = ((DoubleIfVertex) vertex).getThn();
+            els = ((DoubleIfVertex) vertex).getEls();
+        } else {
+            return Optional.empty();
+        }
+
+        StringBuilder builder = new StringBuilder();
+
+        if (includeBrackets) {
+            builder.append("(");
+        }
+        builder.append(generateDescription(predicate, true, includeBrackets));
+        builder.append(" ? ");
+        builder.append(generateDescription(thn, true, includeBrackets));
+        builder.append(" : ");
+        builder.append(generateDescription(els, true, includeBrackets));
+        if (includeBrackets) {
+            builder.append(")");
+        }
+
+        return Optional.of(builder.toString());
+    }
+
+    private Optional<String> tryCreateDescriptionFromSaveLoadAnnotations(Vertex vertex, boolean includeBrackets) {
+        Optional<Constructor<?>> vertexConstructor = Arrays.stream(vertex.getClass().getConstructors())
+            .filter(constructor -> Arrays.stream(constructor.getParameterAnnotations())
+                .anyMatch(annotationsList -> annotationsList.length > 0))
```

Comment:
Potentially dangerous I guess given you're not checking for the types?

Instead of using the constructor, could you just use the parameters that we save?